### PR TITLE
[G-API] Fix streaming kernels in standalone 

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
Xforce_builders_only=linux,docs
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2020.3.0:16.04
build_image:Custom Win=openvino-2020.3.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by TolyaTalamanov** [comment]
@dbudniko Could you have a look ?

**Comment by TolyaTalamanov** [comment]
@rgarnov Could you take a look, please ?

File: modules/gapi/src/backends/streaming/gstreamingbackend.cpp [hunk 1]
```diff
@@ -216,6 +253,21 @@ void cv::gimpl::BGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
...
+
+#else
+
+cv::gapi::GKernelPackage cv::gapi::streaming::kernels()
+{
+    // Still provide this symbol to avoid linking issues
+    util::throw_error(std::runtime_error("cv::gapi::streaming::kernels() isn't supported in standalone"));
+}
+
+#endif //
```
Comment (rgarnov) [inline modules/gapi/src/backends/streaming/gstreamingbackend.cpp:269]: // !defined(GAPI_STANDALONE)

**Review by rgarnov — CHANGES_REQUESTED** [review]

File: modules/gapi/src/backends/streaming/gstreamingbackend.cpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/src/backends/streaming/gstreamingbackend.cpp]: Fixed

File: modules/gapi/src/executor/gstreamingexecutor.cpp [hunk 2]
```diff
@@ -1005,12 +1005,10 @@ cv::gimpl::GStreamingExecutor::GStreamingExecutor(std::unique_ptr<ade::Graph> &&
                     // In the current implementation, such islands
                     // _must_ start with copy
                     GAPI_Assert(isl->in_ops().size() == 1u);
-#if !defined(GAPI_STANDALONE)
```
Comment (TolyaTalamanov) [inline modules/gapi/src/executor/gstreamingexecutor.cpp:1008]: @dmatveev There's no need for that anymore, is there ?

**Review by rgarnov — APPROVED** [review]

File: modules/gapi/src/backends/streaming/gstreamingbackend.cpp [hunk 3]
```diff
@@ -216,6 +253,21 @@ void cv::gimpl::BGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
...
+#else
+
+cv::gapi::GKernelPackage cv::gapi::streaming::kernels()
+{
+    // Still provide this symbol to avoid linking issues
+    util::throw_error(std::runtime_error("cv::gapi::streaming::kernels() isn't supported in standalone"));
+}
+
+#endif // !defined(GAPI_STANDALONE)
+
```
Comment (rgarnov) [inline modules/gapi/src/backends/streaming/gstreamingbackend.cpp:270]: Please add streaming backend sources to the list of source files which are compiled in standalone

**Review by rgarnov — CHANGES_REQUESTED** [review]

File: modules/gapi/src/backends/streaming/gstreamingbackend.cpp (hunk 3 above)
Comment (rgarnov) [reply modules/gapi/src/backends/streaming/gstreamingbackend.cpp:270]: And you'll need to enclose <imgproc.hpp> include with standalone ifdefs

Comment (TolyaTalamanov) [reply modules/gapi/src/backends/streaming/gstreamingbackend.cpp:273]: Done

**Review by rgarnov — APPROVED** [review]

File: modules/gapi/src/executor/gstreamingexecutor.cpp (hunk 2 above)
Comment (dmatveev) [reply modules/gapi/src/executor/gstreamingexecutor.cpp:1008]: I agree

**Pull request merged**